cogito/protocol.go

```diff
@@ -212,18 +217,47 @@ func (src *Source) UnmarshalJSON(data []byte) error {
 // Validate verifies the Source configuration and applies defaults.
 func (src *Source) Validate() error {
 	//
-	// Validate mandatory fields.
+	// Evaluate mandatory fields.
 	//
 	var mandatory []string
-	if src.Owner == "" {
-		mandatory = append(mandatory, "owner")
+	githubRequired := map[string]string{
+		"Owner":       "owner",
+		"Repo":        "repo",
+		"AccessToken": "access_token",
+	}
+
+	defaultSinks := []string{"gchat", "github"}
+	defaultSinksSet := sets.From(defaultSinks...)
+	sinksSet := sets.From(src.Sinks...)
+
+	if sinksSet.Size() > 0 {
+		sinksNotValid := sinksSet.Difference(defaultSinksSet)
```

Comment:
in my understanding this code segment checks if the configured sinks are allowed.

So for me this is not defaultSinks its allowedSinks (or supported).

Additionally, at how many places we will need a list of the defaultSinks? 

This should be a global variable (`[]string{"gchat", "github"}`) which we change once and reuse.

Reply:
Something similar as I did in `vars_test,go`? I thought about it, so we will have just a centralized point to declare the supported sinks.